src/Analysis/Ast/Impl/Types/PythonClassType.cs

```diff
@@ -108,7 +117,7 @@ internal PythonClassType(string name, Location location)
                     return _documentation;
                 }
                 // Make sure we do not cycle through bases back here.
-                if (!Push(this)) {
+                if (!_memberGuard.Push(this)) {
```

Comment:
Technically this can also be called during analysis (like user hovers over a method or variable) to get the doc and this will be another thread.